Forum: Tweak 0032881: Reset Limited Thread View Message Box

fhelfer opened the pull request:
This fixes https://mantis.ilias.de/view.php?id=32881

fhelfer committed:
use KS Message Box

fhelfer committed:
added descriptions reset_limited_view forum

mjansenDatabay commented on Modules/Forum/classes/class.ilObjForumGUI.php:
```diff
@@ -1315,8 +1314,9 @@ protected function selectPostObject() : void
         );
-        $this->showResetLimitedViewInfo();
-        $this->viewThreadObject();
+        $info = $this->getResetLimitedViewInfo();
```
IMO the following condition is missing here:

```php
if ($this->selectedSorting === ilForumProperties::VIEW_TREE && ($this->selected_post_storage->get($thr_pk) > 0)) {
}
```

fhelfer committed:
added condition to limited thread view

mjansenDatabay reviewed (commented):
One (hopefully) last question: Why did you have to use `$this->tpl->setContent(...)` for all these occurances of `$this->viewThreadObject()`?

fhelfer commented:
> One (hopefully) last question: Why did you have to use `$this->tpl->setContent(...)` for all these occurances of `$this->viewThreadObject()`?

viewResetLimitedViewInfo() added the message box via $this->tpl->setOnScreenMessage()
I used the KS Message Box. $this->tpl->setOnScreenMessage() would have added another box around the box.
So i changed the method to return the html string and merged it with viewThreadObject() html
Threrefore i changed all occurences

mjansenDatabay commented:
Yes, but can't you just concatenate the two strings in `viewThreadObject` at the bottom of the function?

fhelfer commented:
Yes but i had to either pass it as a parameter
or call getResetLimitedViewInfo() inside
First option changes the the meaning of the function for only one occurence
Second option sounds good, but i didnt know when (in which case) i had to call it, to ignore it for all other calls

fhelfer committed:
revert viewThreadObject

fhelfer committed:
added thr_pk parameter to contructor

mjansenDatabay commented on Modules/Forum/classes/class.ilObjForumGUI.php:
```diff
@@ -115,8 +115,7 @@ public function __construct($data, int $id = 0, bool $call_by_reference = true,
             );
-
-            $this->showResetLimitedViewInfo();
+            $this->ctrl->setParameter($this, 'thr_pk', $this->objCurrentTopic->getId());
```
Do we really need this line? This seems to be duplicated in https://github.com/ILIAS-eLearning/ILIAS/pull/4672/files#diff-fd85924abce90fbeedabb3e3c4f121dfce9fedd4820943703ccf4a70ab4c31f9R5897

fhelfer replied:
Deleting this line will result in error when returning from limited view
with message Could not find first posting by id: 0
